File: js/task2/S2_ARR_03.js
function chunk(arr, size) {
  if (size <= 0) return null;

  const result = [];

  for (let i = 0; i < arr.length; i += size) {
      result.push(arr.slice(i, i + size));
  }

  return result;
}

console.log(chunk([1, 2, 3, 4, 5], 2));     
console.log(chunk([1, 2, 3, 4, 5], 3));    
console.log(chunk([1, 2, 3], 1));          
console.log(chunk([1, 2, 3], 5));         
console.log(chunk([], 3));                 
console.log(chunk([1, 2, 3], 0));         
console.log(chunk([1, 2, 3], -1));         
console.log(chunk([1, 2, 3], 1.5));

File: python/task2/S2_ARR_03.py
def chunk(arr, size):
  if size <= 0:
      return None

  result = []
  i = 0

  while i < len(arr):
      result.append(arr[i:i+size])
      i += size

  return result

print(chunk([1, 2, 3, 4, 5], 2))
print(chunk([1, 2, 3, 4, 5], 3))
print(chunk([1, 2, 3], 1))
print(chunk([1, 2, 3], 5))
print(chunk([], 3))
print(chunk([1, 2, 3], 0))
print(chunk([1, 2, 3], -1))
print(chunk([1, 2, 3], 1.5))
